pkg/roachprod/roachprod.go

```diff
@@ -777,8 +777,8 @@ func UpdateTargets(
 func updatePrometheusTargets(ctx context.Context, l *logger.Logger, c *install.SyncedCluster) {
 	nodeIPPorts := make([]string, len(c.Nodes))
 	var wg sync.WaitGroup
-	for i, node := range c.VMs {
-		if node.Provider == gce.ProviderName {
+	for i, node := range c.Nodes {
+		if c.VMs[node].Provider == gce.ProviderName {
```

Comment:
I don't think `VMs` is 1 based indexed like nodes is. `VMs[node-1]` instead?

Reply:
Yeah `acceptance/mismatched-locality` nil panics for me. It might accidentally "work" on PCR tests because of the extra workload node?

Reply:
Done.